pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1366,5 +1388,80 @@ public void markBatchMessagePublished() {
         this.hasBatchMessagePublished = true;
     }
 
+    public RateLimiter getDispatchRateLimiter() {
+        return dispatchRateLimiter;
+    }
+    
+    public DispatchRateType getDispatchRateType() {
+        return dispatchRateType;
+    }
+
+    public boolean tryDispatchPermit(long permits) {
+        return permits <= 0 || dispatchRateLimiter == null
+        // acquiring permits must be < configured msg-rate;
+                || dispatchRateLimiter.tryAcquire(permits);
+    }
+    
+    public boolean hasMessageDispatchPermit() {
+        return dispatchRateLimiter == null || dispatchRateLimiter.getAvailablePermits() > 0;
+    }
+
+    private void updateDispatchRate(LocalZooKeeperCacheService localZKCache) {
+        final String path = joinPath(LOCAL_POLICIES_ROOT, DestinationName.get(this.topic).getNamespace());
+        try {
+            Optional<LocalPolicies> policies = localZKCache.policiesCache().getAsync(path).get(cacheTimeOutInSec,
+                    SECONDS);
+            if (policies.isPresent() && policies.get().dispatchRate != null
+                    && policies.get().dispatchRate.topicDispatchRate > 0) {
+                updateDispatchRate(policies.get().dispatchRate);
+                log.info("[{}] configured message-dispatch rate configured at policy {}}", this.topic,
+                        policies.get().dispatchRate);
+                return;
+            }
+        } catch (Exception e) {
+            log.warn("Failed to get message-rate for {}", this.topic, e);
+        }
+        DispatchRate dispatchRate = brokerService.pulsar().getConfiguration().getDispatchRatePerTopicInMsg() > 0
```

Comment:
> Actually, that could just be done inside DispatchRate class

Actually, using admin-api, we always define `DispatchRateType=msgRate/byteRate` and throttling-value in `DispatchRate` class.
However, for dynamic configuration we can't store complex-type so, created two variables `dispatchRatePerTopicInMsg` and `dispatchRatePerTopicInByte`, therefore, we need this logic when we have value set for both configuration.

> Ideally, if both limits are set, both of them should be applied.

This will require 2 `RateLimiter` objects for every topic which we use across all the subscriptions. Do you think considering both limits would be useful?